pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -365,40 +346,37 @@ func (a Adapter) createOrUpdateComponent(componentExists bool, ei envinfo.EnvSpe
 		return err
 	}
 
-	// Add PVC and Volume Mounts to the podTemplateSpec
-	err = kclient.AddPVCAndVolumeMount(podTemplateSpec, volumeNameToPVCName, containerNameToVolumes)
+	// Get PVC volumes and Volume Mounts
+	containers, pvcVolumes, err := kclient.GetPVCVolAndVolMount(containers, volumeNameToPVCName, containerNameToVolumes)
```

Comment:
can we do the volume mount part while generating the containers?

Reply:
Probably can, but that should be done with GetVolumes refactoring. I have not touched those volume logic in this PR.